crates/forge-runner/src/running.rs

```diff
@@ -366,24 +367,37 @@ fn get_latest_block_number(url: &Url) -> Result<BlockId> {
     }
 }
 
-fn get_all_execution_resources(runtime: &ForgeRuntime) -> ExecutionResources {
-    let test_used_resources = &runtime.extended_runtime.extended_runtime.child.resources;
+fn get_all_execution_resources(runtime: ForgeRuntime) -> UsedResources {
+    let mut all_resources = UsedResources {
+        execution_resources: runtime
+            .extended_runtime
+            .extended_runtime
+            .child
+            .resources
+            .clone(),
+        // we construct CallInfo with the `l2_to_l1_messages` field to use
+        // `get_sorted_l2_to_l1_payloads_length` method
+        l2_to_l1_payloads_length: CallInfo {
+            execution: CallExecution {
+                l2_to_l1_messages: runtime
+                    .extended_runtime
+                    .extended_runtime
+                    .child
+                    .l2_to_l1_messages,
+                ..Default::default()
+            },
+            ..Default::default()
+        }
+        .get_sorted_l2_to_l1_payloads_length()
+        .unwrap(),
```

Comment:
Maybe it's just me, but I find it hard comprehend such nesting. Defining a complex variable over multiple lines, with the use of helper variables seems to be more readable.
```rust
    let resources = runtime
        .extended_runtime
        .extended_runtime
        .child
        .resources
        .clone();

    let l2_to_l1_msgs = runtime
        .extended_runtime
        .extended_runtime
        .child
        .l2_to_l1_messages;

    // we construct CallInfo with the `l2_to_l1_messages` field to use
    // `get_sorted_l2_to_l1_payloads_length` method
    let call_info = CallInfo {
        execution: CallExecution {
            l2_to_l1_messages: l2_to_l1_msgs,
            ..Default::default()
        },
        ..Default::default()
    };

    let sorted_l2_to_l1_payloads_length = call_info.get_sorted_l2_to_l1_payloads_length().unwrap();
    
    let mut all_resources = UsedResources {
        execution_resources: resources,
        l2_to_l1_payloads_length: sorted_l2_to_l1_payloads_length,
    };
```

Reply:
Agreed, I hope to work on removing this nesting later as I have some ideas how to tackle it.